src/main/java/com/softserve/teamproject/entity/Student.java

```diff
@@ -111,7 +111,7 @@ public boolean equals(Object otherObject) {
 
   @Override
   public int hashCode() {
-    return Objects.hash(firstName, lastName, group.getId());
+    return Objects.hash(firstName, lastName);
```

Comment:
if you still want to use group, you can make like (group!=null?group.getName:null)